ec2/helpers.go

```diff
@@ -1,63 +1,59 @@
 package ec2
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
+	"fmt"
+
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 )
 
-func getEC2InstanceInfo(context ec2iface.EC2API, instances ...string) (output []*ec2.DescribeInstancesOutput, err error) {
-
-	instanceInput := &ec2.DescribeInstancesInput{
-		InstanceIds: aws.StringSlice(instances),
+func getEC2InstanceInfo(client ec2iface.EC2API, instances []*string) (output []*ec2.Instance, err error) {
+	// Set up our DI input object
+	diInput := &ec2.DescribeInstancesInput{
+		InstanceIds: instances,
 	}
 
-	results, err := context.DescribeInstances(instanceInput)
-	if err != nil {
-		return nil, err
-	}
-	output = append(output, results)
+	// Fetch all the instances described
+	if err = client.DescribeInstancesPages(
```

Comment:
nit: This is a lot shoved into an if conditional. Specifically the anonymous function. Instead of an anonymous function you can just define it with a name within this function and pass it to the client by name:
```go
describeInstancePager := func(page *ec2.DescribeInstancesOutput, lastPage bool) bool {
        ...
}
if err = client.DescribeInstancesPages(diInput, describeInstancePager); err != nil {
        return nil, fmt.Errof("Could not describe EC2 instances\n%v", err)
}
```